pypiserver/__main__.py

```diff
@@ -235,7 +243,7 @@ def main(argv=None):
             c.password_file = v
         elif k in ("-o", "--overwrite"):
             c.overwrite = True
-        elif k in ("--hash-algo"):
+        elif k == "--hash-algo":
```

Comment:
The reason of this change is that `-h` was not working due to the fact that `("--hash-algo")` is equivalent to `"--hash-algo"` which contains `"-h"``.